server/src/test/java/com/jinyframework/keva/server/ServerTest.java

```diff
@@ -56,8 +62,109 @@ static void startServer() throws Exception {
+            assertEquals("1", setAbc);
+        } catch (Exception e) {
+            fail(e);
+        }
+
+        final List<Callable<Object>> tasks = new ArrayList<>();
+        final ExecutorService executor = Executors.newFixedThreadPool(2);
+        final int taskNum = 2;
+        for (int i = 0; i < taskNum; i++) {
+            final int finalI = i;
+            tasks.add(() ->
+                      {
+                          System.out.println("task: " + finalI);
+                          return client.exchange("get abc");
+                      });
+        }
+        try {
+            final List<Future<Object>> futures = executor.invokeAll(tasks, 2, TimeUnit.SECONDS);
+            assertFalse(futures.isEmpty());
+            assertEquals(taskNum, futures.size());
+            for (Future<Object> future : futures) {
+                assertEquals("123", future.get().toString());
+            }
+
+            System.out.println(futures.size());
+        } catch (Exception e) {
+            fail(e);
+        }
+    }
+
+    @Test
+    @Timeout(5)
+    void getSetLongString() {
+        try {
+            final int aKB = 1024;
+            String testStr = "a".repeat(Math.max(0, aKB));
+            String setAbc = client.exchange("set abc " + testStr);
+            String getAbc = client.exchange("get abc");
+            assertEquals("1", setAbc);
+            assertEquals(testStr, getAbc);
+
+            final int aMB = 1024 * aKB;
+            testStr = "a".repeat(Math.max(0, aMB));
+            setAbc = client.exchange("set abc " + testStr);
+            getAbc = client.exchange("get abc");
+            assertEquals("1", setAbc);
+            assertEquals(testStr, getAbc);
+
+            final int MB4 = 4 * aMB;
+            testStr = "a".repeat(Math.max(0, MB4));
+            setAbc = client.exchange("set abc " + testStr);
+            getAbc = client.exchange("get abc");
+            assertEquals("1", setAbc);
+            assertEquals(testStr, getAbc);
+        } catch (Exception e) {
+            fail(e);
+        }
+    }
+
+    @Test
+    @Timeout(20)
+    void getSetLongerString() {
+        try {
+            final int aMB = 1024 * 1024;
+            final int MB8 = 8 * aMB;
+            String testStr = "a".repeat(Math.max(0, MB8));
+            String setAbc = client.exchange("set abc " + testStr);
+            String getAbc = client.exchange("get abc");
+            assertEquals("1", setAbc);
+            assertEquals(testStr, getAbc);
+
+            final int MB16 = 16 * aMB;
+            testStr = "a".repeat(Math.max(0, MB16));
+            setAbc = client.exchange("set abc " + testStr);
+            getAbc = client.exchange("get abc");
+            assertEquals("1", setAbc);
+            assertEquals(testStr, getAbc);
+        } catch (Exception e) {
+            fail(e);
+        }
+    }
+
+    @Test
+    @Timeout(5)
+    void getSetLongKeyString() {
+        try {
+            final int aKB = 1023;
```

Comment:
@tuhuynh27 try this on your machine with 1024

Reply:
Should ignore it as I'll replace NoHeapDB Engine with Chronicle Map soon